js/viewer/annotation/text.js

```diff
@@ -0,0 +1,425 @@
+    var divCont = foreignObj.childNodes[0];
+    if(divCont == null) return;
+    var textInput = divCont.childNodes[0];
+    textInput.style.fontSize = this._ratio * fontSize + "px";
+    textInput.style.height = "1px";
+    textInput.style.height = textInput.scrollHeight + "px";
+}
+
+/**
+ * This function is used to enable drag functionality on the text input
+ */
+Text.prototype.initDragElement = function () {
+
+    pos1 = 0, pos2 = 0, pos3 = 0, pos4 = 0;
+
+    var foreignObj = this.getForeignObj();
+    var divCont = foreignObj.childNodes[0];    
+    var elmnt = null;
+    var currentZIndex = 100;
+    var _self = this;
+    divCont.onmousedown = function() {
+        this.style.zIndex = "" + ++currentZIndex;
+    };
+
+    divCont.onmousedown = dragMouseDown;
+    divCont.onpointerdown = dragMouseDown;
+
+    // This function is called when the user clicks to start dragging the element
+    function dragMouseDown(e) {
+        e.stopImmediatePropagation();
+        elmnt = e.target;
+        elmnt.style.zIndex = "" + ++currentZIndex;
+
+        e = e || window.event;
+        // get the mouse cursor position at startup:
+        pos3 = e.clientX;
+        pos4 = e.clientY;
+        divCont.onmouseup = closeDragElement;
+        divCont.onpointerup = closeDragElement;
+        divCont.onmouseleave = closeDragElement;
+        // call a function whenever the cursor moves:
+        document.onmousemove = elementDrag;
+    }
+
+    // This function is called when the user drags the element
+    function elementDrag(e) {
+        if (!elmnt) {
+        return;
+        }
+
+        e = e || window.event;
+        // Calculate the new cursor position:
+        pos1 = pos3 - e.clientX;
+        pos2 = pos4 - e.clientY;
+        pos3 = e.clientX;
+        pos4 = e.clientY;
+        // Set the element's new position:
+        elmnt.style.top = elmnt.offsetTop - _self._ratio * pos2 + "px";
+        elmnt.style.left = elmnt.offsetLeft - _self._ratio * pos1 + "px";
+    }
+
+    // This function is called when the user stops dragging the element
+    function closeDragElement(e) {
+        document.onmouseup = null;
+        document.onmousemove = null;
+        document.onpointerup = null;
+    }
+}
+
+/**
+ * This function enables the resizing functionality for the text input
+ */
+Text.prototype.initResizeElement = function () {
+
+    var foreignObj = this.getForeignObj();
+    var divCont = foreignObj.childNodes[0];
+    var element = null;
+    var startX, startY, startWidth, startHeight;
+    
+    bottomright = document.createElement("div");
+    bottomright.className = "resizer-bottomright";
+    divCont.appendChild(bottomright);
+    bottomright.addEventListener("mousedown", initDrag, false);
+    bottomright.parentPopup = divCont;
+    bottomright.style.height = this._attrs["font-size"] + "px";
+    bottomright.style.width = this._attrs["font-size"] + "px";
+    bottomright.style["margin-left"] = this._attrs["font-size"] * 0.125 + "px";
```

Comment:
Why 0.125?

Reply:
Added a comment in the code. To position the resized element exactly on the right border, we need to add a margin of 1/8th of the font size to the left of the element, since the padding of the div is 4 times the stroke width.